test/e2e/util.go

```diff
@@ -2298,6 +2298,60 @@ func waitForDeploymentStatus(c clientset.Interface, ns, deploymentName string, d
 	return err
 }
 
+// waitForDeploymentRollbackCleared waits for given deployment either started rolling back or doesn't need to rollback.
+func waitForDeploymentRollbackCleared(c clientset.Interface, ns, deploymentName string) error {
+	return wait.Poll(poll, 1*time.Minute, func() (bool, error) {
+		deployment, err := c.Extensions().Deployments(ns).Get(deploymentName)
+		if err != nil {
+			return false, err
+		}
+		// Rollback not set or is kicked off
+		if deployment.Spec.RollbackTo == nil {
+			return true, nil
+		}
+		return false, nil
+	})
+}
+
+// waitForDeploymentStatusUpdated waits for the deployment's status to match the given status fields. The input int* fields will be ignored if it's nil.
+func waitForDeploymentStatusUpdated(c clientset.Interface, ns, deploymentName string, replicas, updatedReplicas, availableReplicas, unavailableReplicas *int) error {
+	return wait.Poll(poll, 5*time.Minute, func() (bool, error) {
+		deployment, err := c.Extensions().Deployments(ns).Get(deploymentName)
+		if err != nil {
+			return false, err
+		}
+		if replicas != nil && deployment.Status.Replicas != *replicas ||
```

Comment:
I agree with @nikhiljindal. Why do the ints need to be pointers?

Also, why wasn't waitForDeploymentStatus sufficient?